File: scenarios/cosmos-db-typescript-crud/golden/src/cosmos-crud.ts
import { CosmosClient, type SqlQuerySpec } from "@azure/cosmos";

declare const process: {
  env: Record<string, string | undefined>;
  exitCode?: number;
};

interface InventoryItem {
  id: string;
  category: string;
  name: string;
  quantity: number;
}

function requireEnvironment(name: string): string {
  const value = process.env[name];
  if (!value) {
    throw new Error(`Set ${name} before running the application.`);
  }
  return value;
}

function isCosmosError(error: unknown): error is Error & { code: number } {
  return (
    error instanceof Error &&
    "code" in error &&
    typeof error.code === "number"
  );
}

async function main(): Promise<void> {
  const client = new CosmosClient({
    endpoint: requireEnvironment("COSMOS_ENDPOINT"),
    key: requireEnvironment("COSMOS_KEY"),
  });

  try {
    const { database } = await client.databases.createIfNotExists({
      id: "TestDB",
    });
    const { container } = await database.containers.createIfNotExists({
      id: "Items",
      partitionKey: { paths: ["/category"] },
    });

    const item: InventoryItem = {
      id: "item-1",
      category: "electronics",
      name: "Laptop",
      quantity: 1,
    };
    await container.items.create(item);

    const { resource: readItem } = await container
      .item(item.id, item.category)
      .read<InventoryItem>();
    console.log(`Read ${readItem?.name}`);

    const query: SqlQuerySpec = {
      query: "SELECT * FROM c WHERE c.category = @category",
      parameters: [{ name: "@category", value: "electronics" }],
    };
    const iterator = container.items.query<InventoryItem>(query);
    while (iterator.hasMoreResults()) {
      const { resources } = await iterator.fetchNext();
      for (const result of resources) {
        console.log(`Queried ${result.name}`);
      }
    }

    item.quantity = 2;
    await container.item(item.id, item.category).replace(item);
    await container.item(item.id, item.category).delete();
  } catch (error: unknown) {
    if (isCosmosError(error)) {
      console.error(`Cosmos DB request failed with status ${error.code}.`);
    } else {
      throw error;
    }
    process.exitCode = 1;
  } finally {
    client.dispose();
  }
}

await main();

File: scenarios/cosmos-db-java-crud/golden/src/main/java/com/example/CosmosCrud.java
package com.example;

import com.azure.cosmos.CosmosClient;
import com.azure.cosmos.CosmosClientBuilder;
import com.azure.cosmos.CosmosContainer;
import com.azure.cosmos.CosmosDatabase;
import com.azure.cosmos.CosmosException;
import com.azure.cosmos.models.CosmosItemRequestOptions;
import com.azure.cosmos.models.CosmosItemResponse;
import com.azure.cosmos.models.CosmosQueryRequestOptions;
import com.azure.cosmos.models.PartitionKey;
import com.azure.cosmos.models.SqlParameter;
import com.azure.cosmos.models.SqlQuerySpec;
import com.azure.cosmos.util.CosmosPagedIterable;
import java.util.List;

public final class CosmosCrud {
    private CosmosCrud() {
    }

    public static void main(String[] args) {
        String endpoint = requireEnvironment("COSMOS_ENDPOINT");
        String key = requireEnvironment("COSMOS_KEY");

        try (CosmosClient client = new CosmosClientBuilder()
                .endpoint(endpoint)
                .key(key)
                .buildClient()) {
            client.createDatabaseIfNotExists("TestDB");
            CosmosDatabase database = client.getDatabase("TestDB");
            database.createContainerIfNotExists("Items", "/category");
            CosmosContainer container = database.getContainer("Items");

            InventoryItem item =
                    new InventoryItem("item-1", "electronics", "Laptop", 1);
            container.createItem(
                    item,
                    new PartitionKey(item.getCategory()),
                    new CosmosItemRequestOptions());

            CosmosItemResponse<InventoryItem> readResponse =
                    container.readItem(
                            item.getId(),
                            new PartitionKey(item.getCategory()),
                            InventoryItem.class);
            System.out.println("Read " + readResponse.getItem().getName());

            SqlQuerySpec query = new SqlQuerySpec(
                    "SELECT * FROM c WHERE c.category = @category",
                    List.of(new SqlParameter("@category", "electronics")));
            CosmosPagedIterable<InventoryItem> results =
                    container.queryItems(
                            query,
                            new CosmosQueryRequestOptions(),
                            InventoryItem.class);
            results.forEach(result ->
                    System.out.println("Queried " + result.getName()));

            item.setQuantity(2);
            container.replaceItem(
                    item,
                    item.getId(),
                    new PartitionKey(item.getCategory()),
                    new CosmosItemRequestOptions());
            container.deleteItem(
                    item.getId(),
                    new PartitionKey(item.getCategory()),
                    new CosmosItemRequestOptions());
        } catch (CosmosException exception) {
            System.err.printf(
                    "Cosmos DB request failed with status %d: %s%n",
                    exception.getStatusCode(),
                    exception.getMessage());
            System.exit(1);
        }
    }
    private static String requireEnvironment(String name) {
        String value = System.getenv(name);
        if (value == null || value.isBlank()) {
            throw new IllegalStateException("Set " + name + " before running.");
        }
        return value;
    }

    public static final class InventoryItem {
        private String id;
        private String category;
        private String name;
        private int quantity;

        public InventoryItem() {
        }

        public InventoryItem(
                String id,
                String category,
                String name,
                int quantity) {
            this.id = id;
            this.category = category;
            this.name = name;
            this.quantity = quantity;
        }

        public String getId() {
            return id;
        }

        public void setId(String id) {
            this.id = id;
        }

        public String getCategory() {
            return category;
        }

        public void setCategory(String category) {
            this.category = category;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public int getQuantity() {
            return quantity;
        }

        public void setQuantity(int quantity) {
            this.quantity = quantity;
        }
    }
}
